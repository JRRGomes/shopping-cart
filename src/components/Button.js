import styled from 'styled-components';

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.softBlue};
  color: ${({ theme }) => theme.colors.backgroundColor};
  min-width: ${({ theme }) => theme.space[7]};
  height: ${({ theme }) => theme.space[5]};
  padding: ${({ theme }) => theme.space[1]};
  border-radius: ${({ theme }) => theme.radii[2]};
  margin: ${({ theme }) => theme.space[0]};

  &:disabled {
    background-color: ${({ theme }) => theme.colors.lightGrey};
    color: ${({ theme }) => theme.colors.softGrey};
  };

  &:active {
    background-color: ${({ theme }) => theme.colors.darkBlue};
    color: ${({ theme }) => theme.colors.lightBlue};
  };
`
export default Button;
