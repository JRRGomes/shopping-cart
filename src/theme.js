const colors = {
  backgroundColor: '#FBF8F1',
  darkGrey: '#333333',
  softGrey: '#a7a7a7',
  lightGrey: '#dddddd',
  softBlue: '#39AEA9',
  darkBlue: '#557B83',
  lightBlue: '#A2D5AB',
  errorRed: '#FF5677',
};

const space = [
  '0.25rem',
  '0.5rem',
  '0.75rem',
  '1rem',
  '1.5rem',
  '2rem',
  '3rem',
  '4rem',
  '6rem',
  '8rem',
  '12rem',
  '16rem',
  '24rem',
];

const fonts = '"Roboto","Helvetica","Arial",sans-serif';

const fontSizes = [
  '0.25rem',
  '0.5rem',
  '0.75rem',
  '1rem',
  '1.5rem',
  '2rem',
  '3rem',
  '4rem',
  '6rem',
  '8rem',
  '12rem',
  '16rem',
  '24rem',
]

const fontWeights = [0, 300, 400, 500, 600, 700];

const radii = [
  '0.25rem',
  '0.5rem',
  '0.75rem',
  '1rem',
  '1.5rem',
  '2rem',
  '3rem'
]

const zIndex = [
  '0.25rem',
  '0.5rem',
  '0.75rem',
  '1rem',
  '1.5rem',
  '2rem',
  '3rem',
  '4rem',
  '6rem',
  '8rem',
  '12rem',
  '16rem',
  '24rem',
]

const sizes = [
  '0.25rem',
  '0.5rem',
  '0.75rem',
  '1rem',
  '1.5rem',
  '2rem',
  '3rem',
  '4rem',
  '6rem',
  '8rem',
  '12rem',
  '16rem',
  '24rem',
]

const theme = {
  colors,
  space,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  zIndex,
  sizes
};

export default theme
