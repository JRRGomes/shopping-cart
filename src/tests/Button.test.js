import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import { ThemeProvider } from 'styled-components';
import theme from '../theme'

it('renders button correctly', () => {
const button = renderer
.create(<ThemeProvider theme={theme}><Button>my button</Button></ThemeProvider>).toJSON();
expect(button).toMatchSnapshot();
})
