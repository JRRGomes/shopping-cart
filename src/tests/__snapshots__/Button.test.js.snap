// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<button
  className="Button-sc-2h6cpf-0 ijLNHU"
>
  my button
</button>
`;
